Source/Libraries/CorruptCore/CorruptCore.cs

```diff
@@ -745,224 +689,59 @@ public static BlastLayer GenerateBlastLayer(string[] selectedDomains, long overr
-                                    randomAddress = RtcCore.RND.NextLong(0, maxAddress - CachedPrecision);
-
-                                    bus = GetBlastUnits(domain, randomAddress, CachedPrecision, cachedAlignment, cachedEngine);
-                                    if (bus != null)
-                                    {
-                                        bl.Layer.AddRange(bus);
-                                    }
-                                }
-                            }
-
-                            break;
-
-                        case BlastRadius.EVEN: //Evenly distributes the blasts through all selected domains
-
-                            for (var i = 0; i < selectedDomains.Length; i++)
-                            {
-                                domain = selectedDomains[i];
-
-                                for (var j = 0; j < (intensity / selectedDomains.Length); j++)
-                                {
-                                    maxAddress = cachedDomainSizes[i];
-                                    randomAddress = RtcCore.RND.NextLong(0, maxAddress - CachedPrecision);
-
-                                    bus = GetBlastUnits(domain, randomAddress, CachedPrecision, cachedAlignment, cachedEngine);
-                                    if (bus != null)
-                                    {
-                                        bl.Layer.AddRange(bus);
-                                    }
-                                }
-                            }
-
-                            break;
-
-                        case BlastRadius.NONE: //Shouldn't ever happen but handled anyway
-                            return null;
-                    }
+                    var blastInfo = new BlastInfo {
+                        alignment = Alignment,
+                        domainSizes = cachedDomainSizes,
+                        engine = SelectedEngine,
+                        intensity = GetIntensity(overrideIntensity),
+                        precision = CachedPrecision,
+                        selectedDomains = selectedDomains
+                    };
 
+                    var bl = RtcCore.Radius(blastInfo);
                     if (bl.Layer.Count == 0)
                     {
                         return null;
                     }
-                    else
-                    {
-                        return bl;
-                    }
+
+                    return bl;
                 }
                 catch (Exception ex)
                 {
                     var additionalInfo = "";
 
-                    if (MemoryDomains.GetInterface(domain) == null)
-                    {
-                        additionalInfo = "Unable to get an interface to the selected memory domain! \nTry clicking the Auto-Select Domains button to refresh the domains!\n\n";
-                    }
+                    // if (MemoryDomains.GetInterface(domain) == null)
```

Comment:
This is the only spot in the code where I expect a regression. Because the `domain` variable is locally scoped to each `BlastRadiusAlgorithm` function, it's not available for the exception catching here.

We could try to throw a custom exception in each `BlastRadiusAlgorithm` function to provide this information. For now, I'm just removing that info from the exception.